src/mds/MDLog.cc

```diff
@@ -695,20 +726,21 @@ void MDLog::trim(int m)
       new_expiring_segments++;
       expiring_segments.insert(ls);
       expiring_events += ls->num_events;
-      submit_mutex.unlock();
 
       uint64_t last_seq = ls->seq;
       try_expire(ls, op_prio);
+      log_trim_counter.hit();
+      trim_end = ceph::coarse_mono_clock::now();
 
-      submit_mutex.lock();
       p = segments.lower_bound(last_seq + 1);
     }
   }
 
   try_to_commit_open_file_table(get_last_segment_seq());
```

Comment:
@vshankar, sorry, I haven't noticed this earlier. Currently, the function `try_to_commit_open_file_table` expects the `submit_mutex` to be held.

Also, I'm afraid we got a bit too far into the refactoring where we didn't really have to.

Originally we've opened the topic of `submit_mutex` on a line where you wanted to take it in a _new_ code, part of this commit. That's where we argued that it wasn't necessary.
However, we hardly want this PR to refactor the existing code, especially since the other change is doing just that.

This `trim` function isn't new, so we should probably leave its locking pattern as is and avoid unnecessary complications. The other PR addresses the `submit_mutex` usage across the whole class, including this function.

Reply:
That's exactly what Patrick said in this comment here https://github.com/ceph/ceph/pull/52652/files#r1365527610. I've incorrectly marked it as resolved earlier today, so unresolving it back now

Reply:
I missed this comment altogether. Checking now...

Reply:
> @vshankar, sorry, I haven't noticed this earlier. Currently, the function `try_to_commit_open_file_table` expects the `submit_mutex` to be held.
> 
> Also, I'm afraid we got a bit too far into the refactoring where we didn't really have to.
> 
> Originally we've opened the topic of `submit_mutex` on a line where you wanted to take it in a _new_ code, part of this commit. That's where we argued that it wasn't necessary. However, we hardly want this PR to refactor the existing code, especially since the other change is doing just that.
> 
> This `trim` function isn't new, so we should probably leave its locking pattern as is and avoid unnecessary complications. The other PR addresses the `submit_mutex` usage across the whole class, including this function.

ACK. I'll revert the locking changes.

Reply:
Reverted - pushing an update soon.